client/server/pages/index.js

```diff
@@ -131,6 +132,8 @@ function getDefaultContext( request, response, entrypoint = 'entry-main' ) {
 	const reduxStore = createReduxStore( getCachedState( initialReducer, 'root' ) );
 	setStore( reduxStore, getCachedState );
 
+	const queryClient = await createQueryClient();
```

Comment:
This is a good point. `getDefaultContext` should be getting smaller over time rather than bigger, as code is moved to smaller, specialized handlers. Maybe this [`getEnhancedContext`](https://github.com/Automattic/wp-calypso/blob/add/plugins-landing-ssr/client/server/isomorphic-routing/index.js#L66-L77) handler, which runs only for true SSR, would be a better place for creating the `QueryClient`.

On the other hand, constructing the `QueryClient` is a very fast operation that's unlikely to cause any observable slowdown. The [constructor](https://github.com/TanStack/query/blob/900d2c51306a91372ae38d0d6846ae61a165c326/packages/query-core/src/queryClient.ts#L68-L75) merely initializes a few fields to empty-ish values and that's all. No CPU-intensive work and no large memory allocations.

Reply:
Looking deeper, the async part is react-query's [persistQueryClient](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/query-client.ts#L26) which doesn't do anything as long as there's no [window object](https://github.com/TanStack/query/blob/v3.21.1/src/persistQueryClient-experimental/index.ts#L51).
Double-checked and this seems to take `0s` to evaluate.

Reply:
Thanks for catching this.
I moved the `createQueryClient` to `[getEnhancedContext](https://github.com/Automattic/wp-calypso/blob/add/plugins-landing-ssr/client/server/isomorphic-routing/index.js#L66-L77)` as suggested.